### PR TITLE
🌱 Add ginkgo phony in Makefile

### PR DESCRIPTION
**Pull request by kashifest**
we can now do `make ginkgo` to make the ginkgo binary and use it locally

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -252,6 +253,10 @@ $(SETUP_ENVTEST): $(TOOLS_DIR)/go.mod
 	cd $(TOOLS_DIR) && \
 	$(GO) build -tags=tools -o $(BIN_DIR)/setup-envtest sigs.k8s.io/controller-runtime/tools/setup-envtest
 
+.PHONY: $(GINKGO_BIN)
+$(GINKGO_BIN): $(GINKGO) ## Build a local copy of ginkgo.
+
+.PHONY: $(GINKGO)
 $(GINKGO): $(TOOLS_DIR)/go.mod
 	cd $(TOOLS_DIR) && $(GO) get github.com/onsi/ginkgo/v2/ginkgo@$(GINGKO_VER)
 	cd $(TOOLS_DIR) && $(GO) build -tags=tools -o $(BIN_DIR)/ginkgo github.com/onsi/ginkgo/v2/ginkgo
```
Comment (tuminoid) [inline Makefile:262]:
```suggestion
	cd $(TOOLS_DIR) && $(GO) build -tags=tools -o $(BIN_DIR)/$(GINKGO_BIN) github.com/onsi/ginkgo/v2/ginkgo
```

File: Makefile [hunk 2]
```diff
@@ -59,6 +59,7 @@ ENVSUBST_BIN := envsubst
 ENVSUBST := $(TOOLS_BIN_DIR)/$(ENVSUBST_BIN)-drone
 SETUP_ENVTEST = $(TOOLS_BIN_DIR)/setup-envtest
 GINKGO := "$(ROOT_DIR)/$(TOOLS_BIN_DIR)/ginkgo"
+GINKGO_BIN := ginkgo
```
Comment (tuminoid) [inline Makefile:62]: Maybe swap lines 61 and 62 and use `$(GINKGO_BIN)` when defining `GINKGO` ?

Comment (kashifest) [reply Makefile:62]: Yap thats better . Good catch

**Comment by tuminoid** [comment]
Hmm, `make ginkgo`  dirties `hack/tools/go.mod`, and will probably break building other tools.

**Comment by kashifest** [comment]
> Hmm, `make ginkgo` dirties `hack/tools/go.mod`, and will probably break building other tools.

This has been the case for a while now, it is not a side effect of this PR. I will take a look if we can do things differently here.

**Comment by tuminoid** [comment]
> > Hmm, `make ginkgo` dirties `hack/tools/go.mod`, and will probably break building other tools.
> 
> This has been the case for a while now, it is not a side effect of this PR. I will take a look if we can do things differently here.

I thought we got rid of it via #1403

**Comment by kashifest** [comment]
> I thought we got rid of it via #1403

It is still adding the following in go.mod for me

```
  github.com/go-task/slim-sprig v0.0.0-20230315185526-52ccab3ef572 // indirect
  github.com/google/pprof v0.0.0-20210720184732-4bb14d4b1be1 // indirect
```

**Comment by tuminoid** [comment]
> > I thought we got rid of it via #1403
> 
> It is still adding the following in go.mod for me
> 
> ```
>   github.com/go-task/slim-sprig v0.0.0-20230315185526-52ccab3ef572 // indirect
>   github.com/google/pprof v0.0.0-20210720184732-4bb14d4b1be1 // indirect
> ```

So it seems... :disappointed:  

Well kustomize build is happy by luck, so it doesn't again break. We need to figure this out, but we can do it in another PR.

**Review by tuminoid — COMMENTED** [review]
/lgtm

**Comment by tuminoid** [comment]
Created #1413 to follow up.

**Review by lentzi90 — COMMENTED** [review]
/approve

**Pull request merged**